web-server/pages/api/internal/version.ts

```diff
@@ -103,25 +104,61 @@ async function fetchDockerHubTags(): Promise<
   }));
 }
 
+async function fetchLatestGitHubCommit(): Promise<GitHubCommit> {
+  const apiUrl = `https://api.github.com/repos/${githubOrgName}/${githubRepoName}/commits`;
+  const response = await axios.get<GitHubCommit[]>(apiUrl);
+  const latestCommit = response.data[0];
+  return latestCommit;
+}
+
+function isUpdateAvailable(
+  localVersionInfo: ProjectVersionInfo,
+  dockerLatestRemoteTag: TagCompressed,
+  githubLatestCommit: GitHubCommit
+): boolean {
+  const env = process.env.NEXT_PUBLIC_APP_ENVIRONMENT;
+
+  if (env == 'development') {
+    const localBuildDate = new Date(localVersionInfo.current_build_date);
+    const latestRemoteCommitDate = new Date(
+      githubLatestCommit.commit.author.date
+    );
+
+    return latestRemoteCommitDate > localBuildDate;
+  } else {
+    const localBuildDate = new Date(localVersionInfo.current_build_date);
+    const latestRemoteDate = new Date(dockerLatestRemoteTag.last_updated);
+    return latestRemoteDate > localBuildDate;
+  }
+}
+
 async function checkNewImageRelease(): Promise<CheckNewVersionResponse> {
-  const versionInfo = readVersionFile();
-  const localDate = new Date(versionInfo.docker_image_build_date);
-  const remoteTags = await fetchDockerHubTags();
+  const versionInfo = getProjectVersionInfo();
 
-  remoteTags.sort(
+  const dockerRemoteTags = await fetchDockerHubTags();
+  const githubLatestCommit = await fetchLatestGitHubCommit();
+
```

Comment:
this can be combined into a 
```
const [dockerRemoteTags, githubLatestCommit] = await Promise.all([fetchDockerHubTags(), fetchLatestGitHubCommit()])
```

Reply:
thanks